sqlite_utils/db.py

```diff
@@ -1476,6 +1476,21 @@ def create(
             )
         return self
 
+    def duplicate(self, name_new: str) -> "Table":
+        """
+        Duplicate this table in this database.
+
+        :param name_new: Name of new table.
+        """
+        assert self.exists()
+        with self.db.conn:
```

Comment:
I questioned if running a transaction here was the right thing to do - what if the user wants to manage their own transactions? But then I saw that the `.create()` method does the same thing, and since `.duplicate()` is a similar operation to `.create()` I think this is the right approach (consistency is really important).